### ISSUE TITLE
More unexplained @Raw warnings

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#39]
```
What steps will reproduce the problem?

Run the checker on:

import java.util.Set;
import java.util.Arrays;

class AaTest {
  static <X> void test(Set<X>... args) { test2(Arrays.asList(args)); }
  static <X> void test2(Iterable<Set<X>> args) {}
}

What is the expected output? What do you see instead?

Expected: no warnings about @Raw
Actual:

incompatible types.
  static <X> void test(Set<X>... args) { test2(Arrays.asList(args)); }
                                                            ^
  found   : @NonRaw List<@NonRaw Set<@NonRaw X extends @Raw Object>>
  required: @NonRaw Iterable<@NonRaw Set<? extends @NonRaw Object>>


What version of the product are you using? On what operating system?

Linux, v1.0.3a

Please provide any additional information below.

```

Original issue reported on code.google.com by `piotr.zi...@gmail.com` on 10 Dec 2009 at 7:11
